src/Compilers/Core/Private/Microsoft.CodeAnalysis.Private.csproj

```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
+<Project Sdk="Microsoft.NET.Sdk">
+  <!--
+    This project is intended for types that must not be visible outside project `Microsoft.CodeAnalysis`,
+    not even to projects which can access internal types of `Microsoft.CodeAnalysis` (via `InternalsVisibleTo`).
+  -->
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+    <RootNamespace>Microsoft.CodeAnalysis</RootNamespace>
+    <TargetFrameworks>netcoreapp3.1;netstandard2.0</TargetFrameworks>
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
+
+    <!-- NuGet -->
+    <IsPackable>true</IsPackable>
+    <PackageId>Microsoft.CodeAnalysis.Private</PackageId>
```

Comment:
I assumed the DLL needs to be packed and shipped since it's referenced by `Microsoft.CodeAnalysis.Common`?

Reply:
I'm not sure, but my thought was this will just be a small dll included inside Microsoft.CodeAnalysis.Common package.

@RikkiGibson thoughts?

Reply:
Hm, didn't realize I could do that

Reply:
Makes sense to me at a glance but I'm not super familiar with the plan to introduce a new CodeAnalysis dll. That's generally treated as a pretty big change especially if the compiler uses the dll.

Reply:
Good catch @RikkiGibson. I didn't think about that, but the additional dll would not fly with VS/RPS.
@jjonescz Is there an alternative design that doesn't involve a new dll?

Reply:
> Is there an alternative design that doesn't involve a new dll?

Not this simple. Another alternative would be using `[Obsolete]` without `DiagnosticId` and suppress diagnostics via `#pragma`s manually everywhere (lots of places). Or replacing usages of `ISourceGenerator` in Roslyn by `IIncrementalGenerator` everywhere and wrapping `ISourceGenerator`s in an `IIncrementalGenerator` adapter (currently it's the other way around - `IIncrementalGenerator`s are wrapped as `ISourceGenerator`s) - not sure this is even possible, plus it might be a lot of work.

Reply:
Could we use the BCL System.ObsoleteAttribute (to avoid a new DLL), but include `DiagnosticId` on .NET Core only, to allow easier suppression of warnings for .NET Core scenarios?
```csharp
    [Obsolete(Constants.UseIncrementalGenerator
#if NETCOREAPP
        , DiagnosticId = "ROSLYN0001"
#endif
        )]
    public readonly struct GeneratorExecutionContext { ... }
```

Reply:
Yes, but I would still need to add lots of `#pragma warning disable` to build Roslyn itself. That's what I'm trying to avoid here. It's no problem for me to do that, I'm just not sure it would be approved.

Reply:
Maybe we could suppress the obsolete diagnostic for this API globally, instead of doing it in source at every location?

Reply:
Without `DiagnosticId` set in `ObsoleteAttribute`, that could be done only by suppressing all obsolete warnings, I think. That's probably undesirable.